app/models/external_activity.rb

```diff
@@ -113,7 +113,23 @@ class ExternalActivity < ActiveRecord::Base
   # Override the material_type method from SearchModelInterface
   #
   def material_type
-    template_type ? template_type : attributes['material_type']
+    attributes['material_type']
+  end
```

Comment:
You can drop this method now.

Reply:
Do you mean the `material_type` method? 
It needs to override what it gets from
`include SearchModelInterface` which is:

```
  def material_type
    self.class.name.to_s
  end

```

Otherwise none of this works. (Sunspot will index on the ExternalActivity class name.)